src/util/rpc.js

```diff
@@ -1,7 +1,40 @@
-import 'whatwg-fetch';
+import agent from 'superagent';
+
+function toMap(object) {
+  return new Map(
+    Object.keys(object).reduce((arr, key) => [...arr, [key, object[key]]], [])
+  )
+}
+
+function fetch(url, { headers, body, method }) {
+  return new Promise((resolve, reject) => {
```

Comment:
We can use [this library](https://www.npmjs.com/package/superagent-promise) to use superagent in a Promise like way